### PR TITLE
Feature/phone number account creation

### PR DESCRIPTION
**Pull request by plunkettgoogle**
## Issue

- Addresses #701

## Description

Add a phone number field to the account creation page, and save it to the database.

Desktop:
localhost_3000_create-account

Mobile:
localhost_3000_create-account (3)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/addresses technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Create a new account with a phone number set, and check it's set in the database via `select phone_number from user_accounts where email='your@email';`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes
- [ ] I have run `yarn generate:client` if I made backend changes
- [ ] I have exported any new pieces in ui-components

### CODE REVIEW

**Comment by GabeBenjamin** [comment]
Are the screenshots in the description mocks or the actual change? The phone number looks like un-editable text instead of a form field?

**Comment by plunkettgoogle** [comment]
> Are the screenshots in the description mocks or the actual change? The phone number looks like un-editable text instead of a form field?

They're from the actual change. Agree that it's not clear that it's editable. I'm just using the PhoneField component, let me see if there's some styles I can apply to make it look editable.

**Review by anders-schneider — APPROVED** [review]
LGTM!

**Comment by anders-schneider** [comment]
Oh sorry to swoop into an active conversation with my LGTM 😬  Agreed that it looks uneditable in the screenshots, but I found it intuitive enough running it locally (ie it didn't feel "weird" to click into the phone field, and when I did it was clear that it is meant to be edited)

File: sites/public/pages/create-account.tsx [hunk 1]
```diff
@@ -38,10 +39,12 @@ export default () => {
 
   const onSubmit = async (data) => {
     try {
-      const { dob, ...rest } = data
+      const { dob, phoneNumber, ...rest } = data
       await createUser({
         ...rest,
         dob: moment(`${dob.birthYear}-${dob.birthMonth}-${dob.birthDay}`),
+        // Convert (123) 456-7890 to E.164 format with US country code: +11234567890
+        phoneNumber: phoneNumber.replace(/\((\d{3})\) (\d{3})-(\d{4})/, "+1$1$2$3"),
```
Comment (GabeBenjamin) [inline sites/public/pages/create-account.tsx:47]: Does the input component guarantee the phone number will always be in this format?

**Review by GabeBenjamin — APPROVED** [review]
Looks great!

File: sites/public/pages/create-account.tsx (hunk 1 above)
Comment (plunkettgoogle) [reply sites/public/pages/create-account.tsx:47]: From what I could tell, yes.

**Pull request merged**